lte/cloud/go/services/subscriberdb/servicers/protected/indexer_servicer_test.go

```diff
@@ -96,3 +100,128 @@ func encodeIP(ip string) string {
 	ipBytes := net.ParseIP(ip)[12:16] // get just the IPv4 bytes
 	return base64.StdEncoding.EncodeToString(ipBytes)
 }
+
+const gwid1 = "snowflake_ID_1"
+const gwid2 = "snowflake_ID_2"
+
+func TestIndexerSubscriber(t *testing.T) {
+	const (
+		version indexer.Version = 1 // copied from indexer_servicer.go
+	)
+
+	var (
+		types = []string{lte.GatewaySubscriberStateType, lte.MobilitydStateType}
+	)
+
+	subscriberStorage := subscriberdb_test_init.StartTestService(t)
+	idx := indexer.NewRemoteIndexer(subscriberdb.ServiceName, version, types...)
+
+	// define IDs
+	id0 := state_types.ID{Type: lte.GatewaySubscriberStateType, DeviceID: gwid1}
+	id1 := state_types.ID{Type: lte.GatewaySubscriberStateType, DeviceID: gwid2}
+	// define gateway states
+	gatewayState0 := CreateTestGatewaySubscriberState("IMSI001010000000123", "IMSI001010000000456", "IMSI001010000000789", "IMSI001010000001011")
+	gatewayState1 := CreateTestGatewaySubscriberState("IMSI002020000000123", "IMSI002020000000456", "IMSI002020000000789")
+
+	// empty gateway state
+	emptyState := storage.GatewaySubscriberState{Subscribers: map[string]state.ArbitraryJSON{}}
+	emptySerializedState := state_types.SerializedStatesByID{
+		id0: {SerializedReportedState: serialize(t, &emptyState, lte.GatewaySubscriberStateType)},
+	}
+	errs, err := idx.Index("nid0", emptySerializedState)
+	assert.NoError(t, err)
+	assert.Empty(t, errs)
+	foundStates, err := subscriberStorage.GetSubscribersForGateway("nid0", gwid1)
+	assert.NoError(t, err)
+	assert.True(t, cmp.Equal(*foundStates, emptyState))
+
+	// one gateway
+	serializedGatewayState := state_types.SerializedStatesByID{
+		id0: {SerializedReportedState: serialize(t, &gatewayState0, lte.GatewaySubscriberStateType)},
+	}
+	errs, err = idx.Index("nid0", serializedGatewayState)
+	assert.NoError(t, err)
+	assert.Empty(t, errs)
+	foundStates, err = subscriberStorage.GetSubscribersForGateway("nid0", gwid1)
+	assert.NoError(t, err)
+	assert.True(t, cmp.Equal(*foundStates, gatewayState0))
+
+	// two gatewayIDs
+	serializedGatewayState = state_types.SerializedStatesByID{
+		id0: {SerializedReportedState: serialize(t, &gatewayState1, lte.GatewaySubscriberStateType)},
+		id1: {SerializedReportedState: serialize(t, &gatewayState0, lte.GatewaySubscriberStateType)},
+	}
+	errs, err = idx.Index("nid0", serializedGatewayState)
+	assert.NoError(t, err)
+	assert.Empty(t, errs)
+	foundStates, err = subscriberStorage.GetSubscribersForGateway("nid0", gwid1)
+	assert.NoError(t, err)
+	assert.True(t, cmp.Equal(*foundStates, gatewayState1))
+	foundStates, err = subscriberStorage.GetSubscribersForGateway("nid0", gwid2)
+	assert.NoError(t, err)
+	assert.True(t, cmp.Equal(*foundStates, gatewayState0))
+
+	// mixed state (two gateways, two mobilityd states)
```

Comment:
Nitpick: I think it might be nice to have a separate test for the mixed state update.